library/src/otx/backend/openvino/models/segmentation.py

```diff
@@ -137,13 +137,34 @@ def prepare_metric_inputs(
             msg = "The input ground truth masks are not provided."
             raise ValueError(msg)
 
-        return [
-            {
-                "preds": pred_mask,
-                "target": target_mask,
-            }
-            for pred_mask, target_mask in zip(preds.masks, inputs.masks)
-        ]
+        aligned: list[dict[str, Any]] = []
+        for pred_mask, target_mask in zip(preds.masks, inputs.masks):
+            pm = pred_mask
+            tm = target_mask
+
+            # Ensure both are 3D with channel dim first: [1, H, W]
+            # Some backends may return [H, W] — add channel dim if needed.
+            if pm.ndim == 2:
+                pm = pm.unsqueeze(0)
+            if tm.ndim == 2:
+                tm = tm.unsqueeze(0)
+
+            # Align spatial dimensions if swapped (e.g., [1, W, H] vs [1, H, W])
```

Comment:
Why spatial dimensions can be misaligned and swapped? Code here looks as a hack for me. We need to align targets and predictions on the dataset preparation / output preprocessing stage, not here.

Reply:
Let me recheck this after my break. I have made many changes to get training to work, and probably some of these checks can be removed. I do think it's still good to make some of this code more robust against channel orders.